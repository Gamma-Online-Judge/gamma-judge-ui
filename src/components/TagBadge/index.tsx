import { Badge } from "react-bootstrap";

const TagBadge = ({ tag }: TagBadgeProps) => {
    return (
        <Badge pill bg={getBgType(tag)} text={getTextType(tag)} className="mh1">
            {tag}
        </Badge>
    );
}
export default TagBadge;

export type TagBadgeProps = {
    tag: string
}

const getBgType = (tagName: string) => {
    switch (tagName.toLowerCase()) {
        case "dfs":
            return "primary";
        case "grafo":
            return "secondary";
        case "ad hoc":
            return "success";
        case "dp":
            return "danger";
        case "guloso":
            return "warning";
        case "geometria":
            return "info";
        case "matemática":
            return "light";
        default:
            return "dark"
    }
}

const getTextType = (tagName: string) => {
    switch (tagName.toLowerCase()) {
        case "dfs":
            return "light";
        case "grafo":
            return "light";
        case "ad hoc":
            return "light";
        case "dp":
            return "light";
        case "guloso":
            return "dark";
        case "geometria":
            return "light";
        case "matemática":
            return "dark";
        default:
            return "light"
    }
}