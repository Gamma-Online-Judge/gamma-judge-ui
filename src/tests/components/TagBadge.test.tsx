import renderer from 'react-test-renderer';
import TagBadge from '../../components/TagBadge';

test('TagBadge snapshot', async () => {
  const component = renderer.create(<TagBadge tag={''} />);
  let tree = component.toJSON();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
