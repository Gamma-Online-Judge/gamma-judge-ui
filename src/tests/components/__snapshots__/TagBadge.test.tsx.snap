// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagBadge snapshot 1`] = `
<span
  className="mh1 badge rounded-pill bg-primary"
>
  
</span>
`;
